tests/e2e/e2e_test.go

```diff
@@ -12,44 +10,152 @@ import (
 	"github.com/umee-network/umee/v4/tests/grpc"
 )
 
+func (s *IntegrationTestSuite) checkOutflows(umeeAPIEndpoint, denom string, checkWithExcRate bool, amount sdk.Dec, excDenom string) {
+	s.Require().Eventually(
+		func() bool {
+			outflows, err := queryOutflows(umeeAPIEndpoint, denom)
+			s.Require().NoError(err)
+			if checkWithExcRate {
+				outflow := outflows.AmountOf(denom)
+				// get exchange rate
+				exchangeRate, err := queryExchangeRate(umeeAPIEndpoint, excDenom)
+				s.Require().NoError(err)
+				powerReduction := sdk.MustNewDecFromStr("10").Power(6)
+				totalPrice := amount.Quo(powerReduction).Mul(exchangeRate.AmountOf(excDenom))
+				s.T().Log("exchangeRate total price ", totalPrice.String(), "outflow value", outflow.String())
+				oncePer := totalPrice.Quo(sdk.MustNewDecFromStr("100"))
+				threePen := oncePer.Mul(sdk.MustNewDecFromStr("3"))
+				s.Require().True(outflow.GTE(totalPrice.Sub(threePen)) || totalPrice.GTE(outflow.Sub(threePen)))
+			}
+			return outflows.Len() == 1 && outflows[0].Denom == denom
+		},
+		time.Minute,
+		5*time.Second,
+	)
+}
+
+func (s *IntegrationTestSuite) checkSupply(endpoint, ibcDenom string, amount math.Int) {
+	s.Require().Eventually(
+		func() bool {
+			supply, err := queryTotalSupply(endpoint)
+			s.Require().NoError(err)
+			s.Require().Equal(supply.AmountOf(ibcDenom).Int64(), amount.Int64())
+			return supply.AmountOf(ibcDenom).Equal(amount)
+		},
+		time.Minute,
+		5*time.Second,
+	)
+}
+
 func (s *IntegrationTestSuite) TestIBCTokenTransfer() {
+	// s.T().Parallel()
 	var ibcStakeDenom string
 
-	valAddr, err := s.chain.validators[0].keyInfo.GetAddress()
-	s.Require().NoError(err)
-
-	s.Run("send_stake_to_umee", func() {
-		recipient := valAddr.String()
-		token := sdk.NewInt64Coin("stake", 3300000000) // 3300stake
-		s.sendIBC(gaiaChainID, s.chain.id, recipient, token)
+	s.Run("ibc_transfer_quota", func() {
+		// require the recipient account receives the IBC tokens (IBC packets ACKd)
+		gaiaAPIEndpoint := s.gaiaREST()
+		umeeAPIEndpoint := s.umeeREST()
+		// ibc hash of uumee token
+		umeeIBCHash := "ibc/9F53D255F5320A4BE124FF20C29D46406E126CE8A09B00CA8D3CFF7905119728"
+		// ibc hash of uatom token
+		uatomIBCHash := "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2"
+
+		// send uatom from gaia to umee
+		uatomAmount := sdk.NewInt64Coin("uatom", 100000000) // 100ATOM
```

Comment:
how do you know it won't exceed a quota? If price of uatom will go up , then the test will fail. Let's firstly check the price and estimate how much we should send. You can add a todo if needed.